core/src/test/java/org/infinispan/partitionhandling/impl/PreferAvailabilityStrategyTest.java

```diff
@@ -53,162 +73,199 @@
    private PersistentUUIDManagerImpl persistentUUIDManager;
    private AvailabilityStrategyContext context;
    private PreferAvailabilityStrategy strategy;
+   private MockitoSession mockitoSession;
+
+   @DataProvider
+   public static Object[][] conflictResolutionProvider() {
+      return new Object[][]{{RESOLVE}, {IGNORE}};
+   }
 
-   @DataProvider(name = "dp")
-   public Object[][] data() {
-      return new Object[][] {{true},{false}};
+   @Factory(dataProvider = "conflictResolutionProvider")
+   public PreferAvailabilityStrategyTest(ConflictResolution conflicts) {
+      this.conflicts = conflicts;
    }
 
    @BeforeMethod(alwaysRun = true)
    public void setup() {
+      mockitoSession = Mockito.mockitoSession()
+                              .strictness(Strictness.STRICT_STUBS)
```

Comment:
The default is `STRICT_STUBS` in 2.7. Can you clarify what this is doing for us?

Reply:
I thought it was only going to be the default in 3.0, I'll remove it

Reply:
The explicit `MockitoSession` seems to help by allowing me to skip `verify()` for method calls already expected with `when()`. I don't know why, as I thought that was a feature of STRICT_SUBS, but I'll keep it for now.

Reply:
Oh because you are using `noMoreInteracations`. Yeah I guess the Mock must use WARN by default in this case, but the Session default is STRICT_SUBS - kinda odd.